blockchain/v0/reactor.go

```diff
@@ -382,10 +382,17 @@ FOR_LOOP:
 				didProcessCh <- struct{}{}
 			}
 
+			first.DataHash = nil
+			first.Hash()
```

Comment:
Is setting first.DataHash to nil to trigger recomputing the DAHeader?

Reply:
That makes a lot of sense! This was just for debugging purposes, as during validation, an error is thrown because the data availability header's `hash` field is nil due to it not being exported or in [equivalent proto type](https://github.com/lazyledger/lazyledger-core/blob/50f722a510dd2ba8e3d31931c9d83132d6318d4b/proto/tendermint/types/types.proto#L148). 

Do you think we should make a method to force the computation of the hash, or a method to copy the exported `header.DataHash` field into the `DataAvailabilityHeader.hash` field?